subprojects/core-api/src/main/java/org/gradle/api/artifacts/repositories/RepositoryContentDescriptor.java

```diff
@@ -104,15 +105,15 @@ public interface RepositoryContentDescriptor extends InclusiveRepositoryContentD
      *
      * @param configurationNames the names of the configurations the repository will be used for
      */
-    void onlyForConfigurations(String... configurationNames);
+    void onlyForConfigurations(@Nullable String... configurationNames);
```

Comment:
Should this be nullable? I don't think it accepts null, or it's at least not documented what it does.